### PR TITLE
Update all unit-test with @covers annotation, add custom recordedOn test for AggregateChanged, use assertSame where possible, update infection config & badge

### LINKED ISSUE #160: use assertSame for scalar values in our tests
https://github.com/patchlevel/event-sourcing/pull/158#discussion_r777716989

### PR DESCRIPTION
**Pull request by DanielBadura**
fix #160

### CODE REVIEW

**Comment by DanielBadura** [comment]
Not sure what psalm wants here:
```
ERROR: DocblockTypeContradiction - tests/Unit/Aggregate/AggregateRootTest.php:63:15 - Docblock-defined type 1 for $profile->playhead() is never int(2) (see https://psalm.dev/155)
        self::assertSame(2, $profile->playhead());

File: tests/Unit/Aggregate/AggregateChangedTest.php [hunk 1]
```diff
@@ -10,29 +10,31 @@
 use Patchlevel\EventSourcing\Aggregate\AggregateChangeRecordedAlready;
 use Patchlevel\EventSourcing\Tests\Unit\Fixture\Email;
 use Patchlevel\EventSourcing\Tests\Unit\Fixture\ProfileCreated;
+use Patchlevel\EventSourcing\Tests\Unit\Fixture\ProfileCreatedWithCustomRecordedOn;
 use Patchlevel\EventSourcing\Tests\Unit\Fixture\ProfileId;
 use PHPUnit\Framework\TestCase;
 
 use const PHP_VERSION_ID;
 
+/** @covers \Patchlevel\EventSourcing\Aggregate\AggregateChanged */
```
Comment (DavidBadura) [inline tests/Unit/Aggregate/AggregateChangedTest.php:19]: can we use attributes here? :) 

```php
PHPUnit\Framework\Attributes\CoversClass(string $className)
```

https://github.com/sebastianbergmann/phpunit/issues/4502

Comment (DavidBadura) [reply tests/Unit/Aggregate/AggregateChangedTest.php:19]: nvm, phpunit v10 is not released yet :D

**Comment by DavidBadura** [comment]
> Not sure what psalm wants here:
> 
> ```
> ERROR: DocblockTypeContradiction - tests/Unit/Aggregate/AggregateRootTest.php:63:15 - Docblock-defined type 1 for $profile->playhead() is never int(2) (see https://psalm.dev/155)
>         self::assertSame(2, $profile->playhead());
> ```

I don't know, maybe we should add this in the psalm baseline

**Comment by DanielBadura** [comment]
Maybe because `playhead` is `readonly` and already resutled in `1`? I want to check it one more time before adding it to the baseline

**Comment by DavidBadura** [comment]
on the aggregate, playhead is internal not readonly. on the aggregateChange it is readonly, but this is not used here.

**Review by DavidBadura — APPROVED** [review]

**Pull request merged**